### ISSUE TITLE
`set-query!` can cause Om to attempt to focus an old query with a new path

### ISSUE THREAD

**Issue by Peeja** [omcljs/om#779]
Demonstrated [here](https://github.com/Peeja/om/commit/5bfae7e77d1fee073fe21e6924cc239543b6871d), on the devcard `om.devcards.core/test-query-setting`.

The card shows the current user's pets. Clicking a pet in the sidebar selects the pet, and shows more info in the main area. This is accomplished with `set-query!`.

However, when "Jasmine" is clicked, the app throws an error. The error comes [from the indexer](https://github.com/omcljs/om/blob/master/src/main/om/next.cljc#L1733-L1735), which is trying to focus the *old* root query,

```clj
[{:om.devcards.core/route-data
  [{[:app/current-user _]
    [{:user/pets [:db/id :pet/name]}]}
   {[:pet/by-id 0]
    [:db/id :pet/name :pet/favorite-toy]}]}]
```

to the new path, `[:om.devcards.core/route-data [:pet/by-id 1]]`. That is, the query it's focusing has Alfred's ident, `[:pet/by-id 0]`, but the path it's focusing has Jasmine's, `[:pet/by-id 1]`. This leads to an error in `query-template`.

---

It's also entirely possible I'm doing something wrong, in which case I'd be happy to be told what that is. 😃

**Comment by Peeja** [comment]
After further investigation: In `set-query!`, after [this `swap!`](https://github.com/omcljs/om/blob/master/src/main/om/next.cljc#L1348-L1349), `(get-query c)` is correct, where `c` is the `PetsPage` component whose query is being set, because it finds the new query in the `::queries` in the app state. However, `(get-query root)` has the wrong ident, as it has no entry in `::queries`. This is why the subsequent `p/reindex!` fails.

**Comment by anmonteiro** [comment]
fixed 1ce79f114fc602660a1c2e2e09ac4e9d85f8b54f
